client/http_sender.go

```diff
@@ -125,14 +125,11 @@ func (s *httpSender) Send(_ context.Context, call Call) {
 				// the errors we'll sweep up in this net shouldn't be retried,
 				// but we can't really know for sure which.
 				log.Warningf("failed to send HTTP request or read its response: %s", t)
-				return retry.Continue, nil
-			default:
-				// Can't retry in order to recover from this error. Propagate.
-				return retry.Break, err
+				return retry.Continue, err
 			}
 		}
-		// On successful post, we're done with retry loop.
-		return retry.Break, nil
+		// If not previously handled, propagate the error (which may be nil).
```

Comment:
I don't really like this pattern where we've just finished an `if err != nil` block but `err` may still be nil. I'd rather keep the old pattern with a `return retry.Break, err` inside the if, and then outside the if we can use a hard-coded nil.